angular-app/src/app/discovery/discovery.component.html

```diff
@@ -2,38 +2,58 @@
   <span>Discover</span>
 </mat-toolbar>
 
-<mat-selection-list #tags>
-  <mat-list-option *ngFor="let tag of tagOptions">
-    {{tag}}
-  </mat-list-option>
-</mat-selection-list>
+<div class="filter-chip">
+        <mat-chip class="pi-chip-filter"
+                  [matMenuTriggerFor]="longListFilter"
+                  [selected]="selectedLongListTags.length > 0">
+            {{getMultiSelectChipName(selectedLongListTags) || "Search filter"}}
+            <mat-icon matChipRemove
+                      (click)="selectedLongListTags = clearSelectedTags()"
+                      *ngIf="selectedLongListTags.length > 0">cancel</mat-icon>
+            <mat-icon matChipRemove
+                      *ngIf="selectedLongListTags.length === 0"
+                      class="pi-chip-filter-dropdown">arrow_drop_down</mat-icon>
+        </mat-chip>
+
+        <mat-menu #longListFilter="matMenu"
+                  class="pi-chip-filter-menu"
+                  overlapTrigger="false"
+                  xPosition="before"
+                  yPosition="below">
+            <!-- HACK: Stop the mat-menu from closing by stopping event propagation -->
+            <div class="pi-ui mat-typography"
+                 (click)="$event.stopPropagation()">
+                <div class="pi-chip-filter-header">
+                    <div class="pi-chip-filter-title">
+                        Long list filter
```

Comment:
Please update this title to be something more relevant for you app.

Reply:
I changed this name in the discovery-filtering branch which has already been pushed to master, so this will be fixed when I resolve merge conflicts after this PR is approved